pkg/plugins/gateway/util.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2024 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gateway
+
+import (
+	"encoding/json"
+	"slices"
+	"strings"
+
+	configPb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	extProcPb "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	envoyTypePb "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/vllm-project/aibrix/pkg/utils"
+	"k8s.io/klog/v2"
+)
+
+// validateStreamOptions validates whether stream options to include usage is set for user request
+func validateStreamOptions(requestID string, user utils.User, jsonMap map[string]interface{}) *extProcPb.ProcessingResponse {
+	if user.Tpm != 0 {
+		streamOptions, ok := jsonMap["stream_options"].(map[string]interface{})
+		if !ok {
+			klog.ErrorS(nil, "no stream option available", "requestID", requestID, "jsonMap", jsonMap)
+			return generateErrorResponse(envoyTypePb.StatusCode_InternalServerError,
+				[]*configPb.HeaderValueOption{{Header: &configPb.HeaderValue{
+					Key: HeaderErrorNoStreamOptions, RawValue: []byte("stream options not set")}}},
+				"no stream option available")
+		}
+		includeUsage, ok := streamOptions["include_usage"].(bool)
+		if !includeUsage || !ok {
+			klog.ErrorS(nil, "no stream with usage option available", "requestID", requestID, "jsonMap", jsonMap)
+			return generateErrorResponse(envoyTypePb.StatusCode_InternalServerError,
+				[]*configPb.HeaderValueOption{{Header: &configPb.HeaderValue{
+					Key: HeaderErrorStreamOptionsIncludeUsage, RawValue: []byte("include usage for stream options not set")}}},
+				"no stream with usage option available")
+		}
+	}
+	return nil
+}
+
+// getRoutingStrategy retrieves the routing strategy from the headers or environment variable
+// It returns the routing strategy value and whether custom routing strategy is enabled.
+func getRoutingStrategy(headers []*configPb.HeaderValue) (string, bool) {
```

Comment:
rest methods beside `validateStreamOptions` are all from previous change, just refactor to util.go? any change I need to be aware of?

Reply:
Just a refactor, no changes in other functions.